check_api/src/main/java/com/google/errorprone/fixes/SuggestedFixes.java

```diff
@@ -1117,7 +1118,7 @@ private static Tree suppressibleNode(TreePath path) {
                         && ((ClassTree) tree).getSimpleName().length() != 0)
                     // Lambda parameters can't be suppressed unless they have Type decls
                     || (tree instanceof VariableTree
-                        && getStartPosition(((VariableTree) tree).getType()) != -1))
+                        && !hasNoExplicitType((VariableTree) tree, state)))
```

Comment:
This shows that `hasNoExplicitType` should perhaps be renamed to `hasExplicitType`, with negated semantics. WDYT?

Reply:
I agree the double-negation is unfortunate, either `hasExplicitType` with negated semantics, or `hasImplicitType` with the same semantics seem like an improvement. If you want to make one of those changes here I'm happy to approve it :)